ecosystem/indexer-grpc/indexer-transaction-generator/src/script_transaction_generator.rs

```diff
@@ -0,0 +1,264 @@
+// Copyright (c) Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    config::{ScriptStep, ScriptTransactionGeneratorConfig},
+    managed_node::ManagedNode,
+};
+use anyhow::{Context, Result};
+use aptos::{
+    account::fund::FundWithFaucet,
+    common::types::{CliCommand, MovePackageDir, ScriptFunctionArguments, TransactionOptions},
+    governance::CompileScriptFunction,
+    move_tool::{CompileScript, RunScript},
+};
+use aptos_protos::{
+    indexer::v1::{raw_data_client::RawDataClient, GetTransactionsRequest},
+    transaction::v1::Transaction,
+};
+use std::{
+    collections::HashMap,
+    path::{Path, PathBuf},
+};
+
+/// GRPC request metadata key for the token ID.
+const LOCAL_INDEXER_GRPC_URL: &str = "http://127.0.0.1:50051";
+const TRANSACTION_STREAM_TIMEOUT_IN_SECS: u64 = 60;
+const DEFAULT_FUND_AMOUNT_IN_OCTA: u64 = 100_000_000;
+
+impl ScriptTransactionGeneratorConfig {
+    fn validate(&self) -> Result<()> {
+        // Validate the script transactions.
```

Comment:
maybe we can also return early or raise an error if scripted_transactions is empty?

Reply:
done; error out when transactions are empty.